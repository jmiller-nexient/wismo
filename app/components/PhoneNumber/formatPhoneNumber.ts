export default function formatPhoneNumber(value: string) {

    const formattedPhoneNumber = value.length === 7 ? `${value.slice(0, 3)}-${value.slice(3, value.length)}`
        :
        `(${value.slice(0, 3)}) ${value.slice(3, 6)}-${value.slice(6, value.length)}`;

    return formattedPhoneNumber;
}
