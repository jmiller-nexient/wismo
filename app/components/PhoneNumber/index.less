@import '../../styles/themes/index.less';

.dte-wismo-phone-number {
  color: @dte-gray-500;
  font-family: @dte-wismo-font-family;
  font-size: @dte-wismo-font-size-1;
}
