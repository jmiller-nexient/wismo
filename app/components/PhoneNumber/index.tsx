import * as React from 'react';

import formatPhoneNumber from './formatPhoneNumber';
import './index.less';

interface IPhoneNumberProps {
  value: string;
}

const PhoneNumber: React.FC<IPhoneNumberProps> = (props: IPhoneNumberProps) => {
  const { value } = props;

  if (value.length !== 10 && value.length !== 7) {
    throw new Error('phone numbers must be 10 or 7 numeric digits.');
  }

  const formattedPhoneNumber = formatPhoneNumber(value);

  return (
    <span className="dte-wismo-phone-number">
      {formattedPhoneNumber}
    </span>
  );
};

export default PhoneNumber;
