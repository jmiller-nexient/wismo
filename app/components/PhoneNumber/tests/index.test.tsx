import React, { ReactElement } from 'react';
import renderer from 'react-test-renderer';

import PhoneNumber from '..';
import {shallow} from 'enzyme';

describe('<PhoneNumber />', () => {
  it('should render correctly when phone number has 10 characters', () => {
    const tree: ReactElement = renderer.create(
      <PhoneNumber value={'1122334455'} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly when phone number has 7 characters', () => {
    const tree: ReactElement = renderer.create(
      <PhoneNumber value={'1122334'} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should throw an error when phone number is not 10 or 7 characters', () => {
    expect(() => shallow(<PhoneNumber value={'123456'}/>)).toThrowError();
  });

  it('should throw an error when phone number is more than 10 characters', () => {
    expect(() => shallow(<PhoneNumber value={'12345678910'}/>)).toThrowError();
  });
});
