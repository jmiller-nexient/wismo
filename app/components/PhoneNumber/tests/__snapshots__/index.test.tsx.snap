// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneNumber /> should render correctly in case of 7 characters 1`] = `
<span
  className="dte-wismo-phone-number"
>
  112-2334
</span>
`;

exports[`<PhoneNumber /> should render correctly in case of 10 characters 1`] = `
<span
  className="dte-wismo-phone-number"
>
  (112) 233-4455
</span>
`;
